gray/processing.py

```diff
@@ -11,34 +14,102 @@
 log = logging.getLogger(__name__)
 
 
-class FormattingError(Exception):
+class Error(Exception):
+    """Base Error class for Gray."""
     exit_code = 1
 
 
+class ConfigurationError(Error):
+    """Raised on a configuration error."""
+    exit_code = 2
+
+
+class FormattingError(Error):
+    """Raised when a formatting error occured."""
+
+
+def _compile_patterns(exclude: Optional[str], extend: Optional[str]) -> List[re.Pattern]:
+    """Compiles the exclude patterns.
+
+    Args:
+        exclude: The default exclusion regex.
+        extend: Additional regex to exclude.
+    Returns:
+        A list of regex patterns.
+    Raises:
+        ConfigurationError: The regex patterns cannot be compiled.
+    """
+    compiled: List[re.Pattern] = []
+    try:
+        for pattern in (exclude, extend):
+            if pattern:
+                compiled.append(re.compile(pattern))
+    except re.error as error:
+        raise ConfigurationError(f"Invalid regex passed for exclusion: {error}")
+    return compiled
+
+
+def _is_excluded(path: Path, excludes: Optional[Sequence[re.Pattern]]) -> bool:
+    """Returns True if the given path is in the exclusion list.
+
+    Windows paths are treated as posix path to ensure the same exclusion
+    patterns can be used as on other Unix based OSes.
+
+    Args:
+        path: The path to the file or directory to check for exclusion.
+        excludes: Optional list of exclusion patterns.
+    Returns:
+        True if the path matches .
+    """
+    if not excludes:
+        return False
+    posix_path = str(path.as_posix())
+    for exclude in excludes:
+        if exclude.match(posix_path):
+            return True
+    return False
```

Comment:
return any(exclude.match(posix_path) for exclude in excludes)

Reply:
Done